### PR TITLE
CATROID-430 Refactor import of projects activity

### PR DESCRIPTION
**Pull request by DinosaurierRex**
Moved Import logic from ProjectListFragment to new ProjectImportController
Calling Activity gets now notified when Controller finished

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/io/ProjectImportController.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		if (!uriToImport.getScheme().equals("file")) {
+			throw new IllegalArgumentException("importProject has to be called with a file uri. (not a content uri");
+		}
+
+		File cacheFile = new File(CACHE_DIR, CACHED_PROJECT_ZIP_FILE_NAME);
+		if (cacheFile.exists()) {
+			cacheFile.delete();
+		}
+
+		File src = new File(uriToImport.getPath());
```
Comment (PlantinumX) [inline catroid/src/main/java/org/catrobat/catroid/io/ProjectImportController.java]: pls move src before creating the cache file

Comment (DinosaurierRex) [reply catroid/src/main/java/org/catrobat/catroid/io/ProjectImportController.java]: While reviewing we triggered a situation which seems like an endless Loop in the ImportTask. PR now set to WIP

**Comment by PlantinumX** [comment]
Status on this PR,please?

**Comment by DinosaurierRex** [comment]
> Status on this PR,please? -> I see the androidX integration. Should we do a seperate PR for the androidX migration and on top of that this PR

Yes that would probably be a good idea :+1:

**Comment by DinosaurierRex** [comment]
Android X Migration was done already in [PR-3471](https://github.com/Catrobat/Catroid/pull/3471)

**Pull request closed without merging**